src/com/trilead/ssh2/Connection.java

```diff
@@ -137,6 +138,23 @@ public Connection(String hostname, int port)
 	{
 		this.hostname = hostname;
 		this.port = port;
+		this.sourceAddress = null;
+	}
+	
+	/**
+	 * Prepares a fresh <code>Connection</code> object which can then be used
+	 * to establish a connection to the specified SSH-2 server.
+	 * 
+	 * @param hostname
+	 *            the host where we later want to connect to.
+	 * @param port
+	 *            port on the server, normally 22.
+	 */
+	public Connection(String hostname, int port, String sourceAddress)
+	{
+		this.hostname = hostname;
+		this.port = port;
+		this.sourceAddress = sourceAddress;
```

Comment:
You could have just chained the constructors here; it makes future changes simpler, but is no real issue.